lib/ast_ml.ml

```diff
@@ -0,0 +1,134 @@
+module type S_P_TREE = sig
+  module Position : sig
+    (*pos_fname is the file name;
+      pos_lnum is the line number; 
+      pos_bol is the offset of the beginning of the line (number of characters between the beginning of the lexbuf and the beginning of the line); 
+      pos_cnum is the offset of the position (number of characters between the beginning of the lexbuf and the position).  
+    *)
+    type t = {
+      pos_fname : string;
+      pos_lnum : int;
+      pos_bol : int;
+      pos_cnum : int;
+    }
+    type position = (t * t)
+  end
+
+  module Pattern : sig
+    type t =
+      | Left of t
+      | Right of t
+      | LocalPattern of (Location.t * LPattern.t)
+      | Pair of (t * t)
+      | Var of Var.t
+      | Wild
+  end
+  
+  module Choreography : sig
+    type sugaredSend = 
```

Comment:
What are the tradeoffs behind the design decision to separate sugardSend into its own type?